src/green_tree.jl

```diff
@@ -87,11 +87,11 @@ function Base.show(io::IO, ::MIME"text/plain", node::GreenNode, str::AbstractStr
     _show_green_node(io, node, "", 1, str, show_trivia)
 end
 
-function build_tree(::Type{GreenNode}, stream::ParseStream; kws...)
-    build_tree(GreenNode{SyntaxHead}, stream; kws...) do h, srcrange, cs
+# we don't need the `filename` and `first_line` kwargs, but some trees do, so we allow them to be passed
+function build_tree(::Type{GreenNode}, stream::ParseStream; filename=nothing, first_line=nothing)
```

Comment:
It's confusing to know what to do with this part: do we want all `build_tree(::Type{TreeType}, ...)` methods to need to know about each other for all `TreeType`? This seems impossible (eg, external packages) which would imply these should just be an error rather than ignored.

But on the other hand making this an error means `build_tree()` with keywords can't be used generically.

It seems there's two consistent things to do:
* Be generic and ignore unknown keywords (current status quo. not a good user experience, as in #341)
* Throw on unknown keywords (makes `parseall` etc couple the `TreeType` with keywords)

This PR takes an awkward middle path between those two options. Which might be ok but seems inconsistent?

Reply:
Yeah, I guess if we want a more decoupled approach, we could copy Makie and add a `validate_kwargs` function which is called on the tree type and the passed-in kwargs (and throws or returns `nothing`), and is called in `parseall` or `_parse` or something like that. Then custom trees which wish to support additional kwargs would need to define a `validate_kwargs` method to allow them to pass through.

Reply:
That sounds promising! So we define `validate_parse_args(::Type{GreenNode}; kws...)` and users with custom trees can overload it?